src/main/java/org/apache/maven/plugins/shade/mojo/ShadeMojo.java

```diff
@@ -285,6 +286,14 @@
     @Parameter( defaultValue = "false" )
     private boolean generateUniqueDependencyReducedPom;
 
+    /**
+     * Add dependency reduced POM to the JAR instead of the original one provided by the project.
```

Comment:
The documentation should also say:
If {@code createDependencyReducedPom} is {@code false} this parameter will be ignored.

Reply:
LOL, I was just about to push the same change :)